import { AxiosError } from 'axios';
export * from './gen/errors';
export declare class CreateAndUploadFileError extends Error {
    readonly innerError?: AxiosError<unknown, any> | undefined;
    readonly file?: {
        id: string;
        botId: string;
        name: string | null;
        size: number | null;
        contentType: string;
        tags: {
            [k: string]: string;
        };
        createdAt: string;
        updatedAt: string;
        accessPolicies: ("integrations" | "public_content")[];
        index: boolean;
        status: "upload_pending" | "upload_failed" | "upload_completed" | "indexing_pending" | "indexing_failed" | "indexing_completed";
        failedStatusReason?: string | undefined;
        uploadUrl: string;
    } | undefined;
    constructor(message: string, innerError?: AxiosError<unknown, any> | undefined, file?: {
        id: string;
        botId: string;
        name: string | null;
        size: number | null;
        contentType: string;
        tags: {
            [k: string]: string;
        };
        createdAt: string;
        updatedAt: string;
        accessPolicies: ("integrations" | "public_content")[];
        index: boolean;
        status: "upload_pending" | "upload_failed" | "upload_completed" | "indexing_pending" | "indexing_failed" | "indexing_completed";
        failedStatusReason?: string | undefined;
        uploadUrl: string;
    } | undefined);
}
